Basic and token authentication support

josephglanville opened the pull request:
Implements support for "basic" and "token" authentication methods.

Partial implementation of what is called for in #49, to go the next step will need to extract the role from the authentication provider and then use that to authorise actions on the connection.

josephglanville commented:
Similar to #54 I could do with some guidance here on if it's best to have authentication for mqtt enabled independently of the pulsar `authenticationEnabled` setting or not.

codelipenghui reviewed (commented):
The change looks good, could you please also help update the README for how to enable authentication for mop?

codelipenghui commented:
@josephglanville I noticed there are some checkstyle violations, please check, thanks.

zymap commented on mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/support/ProtocolMethodProcessorImpl.java:
```diff
@@ -120,6 +129,34 @@ public void processConnect(Channel channel, MqttConnectMessage msg) {
+                try {
+                    AuthenticationDataSource authData = getAuthData(authMethod, payload);
+                    if (authProvider != null && authData != null) {
+                        authService.getAuthenticationProvider(authMethod)
```
duplicate with line 138?

josephglanville replied:
Yep, vestigial from different approach, will remove.

zymap reviewed (commented):
Nice contribute! Overall looks good

josephglanville commented:
@codelipenghui I have refactored this to not do the configuration in the ChannelInitialiser and rather ensure appropriate authentication providers and methods are configured at start time. I also added support to proxy connections to use authetnication.

Main things to do now are:
Tests for proxy auth
Failure tests (invalid creds etc)
Tests for token authentication.

codelipenghui commented:
@josephglanville Sorry for the late response, I have merged the PR.
